src/buffer.rs

```diff
@@ -685,6 +685,171 @@ mod tests {
     use crate::ffi;
     use crate::Python;
 
+    #[test]
+    fn test_debug() {
+        // PyBuffer {
+        //    buf: 0x00000001012c65d0,
+        //    obj: 0x00000001012c65b0,
+        //    len: 5,
+        //    itemsize: 1,
+        //    readonly: 1,
+        //    ndim: 1,
+        //    format: 0x00000001016747d8,
+        //    shape: 0x0000600001b74010,
+        //    strides: 0x0000600001b74018,
+        //    suboffsets: 0x0000000000000000,
+        //    internal: 0x0000000000000000,
+        //}
+        Python::with_gil(|py| {
+            let bytes = py.eval("b'abcde'", None, None).unwrap();
+            let buffer: PyBuffer<u8> = PyBuffer::get(bytes).unwrap();
+
+            let dbg = format!("{:?}", buffer);
+
+            assert!(dbg.starts_with("PyBuffer"));
+            assert!(dbg.contains("{ buf: "));
+            assert!(dbg.contains(", obj: "));
+            assert!(dbg.contains(", len: 5"));
+            assert!(dbg.contains(", itemsize: 1"));
+            assert!(dbg.contains(", readonly: 1"));
+            assert!(dbg.contains(", ndim: 1"));
+            assert!(dbg.contains(", format: "));
+            assert!(dbg.contains(", shape: "));
+            assert!(dbg.contains(", strides: "));
+            assert!(dbg.contains(", suboffsets: "));
+            assert!(dbg.contains(", internal: "));
```

Comment:
Checking the fragments to avoid the fact the pointers are different each time I guess? I wonder if an alternative could be to replace the pointers with `<some_ptr>` or similar so that you can do a `==` on the whole `dbg` string?

Reply:
Yeah I started with regex but it's only a secondary dependency. Replacing the pointers with known values is a good idea - I'll try that.

Reply:
I refactored to use a big format string to be able to `==` the whole thing. I think this is better but by nature this test may be a little tautological. At least it should catch changes in behavior.